tests/utilities/test_apply_func.py

```diff
@@ -115,24 +150,54 @@ class ModelExample:
         reduced["h"].segment_ids, expected_result["h"].segment_ids
     ), "Reduction of a dataclass did not yield the desired result"
 
-    assert dataclasses.is_dataclass(reduced["i"]) and not isinstance(
-        reduced["i"], type
-    ), "Reduction of a dataclass should result in a dataclass"
-    assert dataclasses.is_dataclass(reduced["i"].feature) and not isinstance(
-        reduced["i"].feature, type
-    ), "Reduction of a nested dataclass should result in a nested dataclass"
-    assert (
-        reduced["i"].example_ids == expected_result["i"].example_ids
-    ), "Reduction of a nested dataclass did not yield the desired result"
+    def _assert_nested_dataclass_reduction(actual: ModelExample, expected: ModelExample):
+
+        assert dataclasses.is_dataclass(actual) and not isinstance(
+            actual, type
+        ), "Reduction of a dataclass should result in a dataclass"
+        assert dataclasses.is_dataclass(actual.feature) and not isinstance(
+            actual.feature, type
+        ), "Reduction of a nested dataclass should result in a nested dataclass"
+        assert (
+            actual.example_ids == expected.example_ids
+        ), "Reduction of a nested dataclass did not yield the desired result"
+        assert torch.allclose(
+            actual.label, expected.label
+        ), "Reduction of a nested dataclass did not yield the desired result"
+        assert torch.allclose(
+            actual.feature.input_ids, expected.feature.input_ids
+        ), "Reduction of a nested dataclass did not yield the desired result"
+        assert np.allclose(
+            actual.feature.segment_ids, expected.feature.segment_ids
+        ), "Reduction of a nested dataclass did not yield the desired result"
```

Comment:
what is the point of these messages? isn't it obvious that something did not yield the desired result when this assertion fails? the function is called `_assert_nested_dataclass_reduction` which should already give a clear idea that we are dealing with nested dataclasses here, and the message is just repeated. it does not add to the readability of the test.

Reply:
Agreed. These assertions preexisted this PR; I just wrapped them in `_assert_nested_dataclass_reduction`. The latest version cuts down on the message duplication.